bindinfo/handle.go

```diff
@@ -254,44 +276,60 @@ func (h *BindHandle) DropBindRecord(record *BindRecord) (err error) {
 	return err
 }
 
-// DropInvalidBindRecord execute the drop bindRecord task.
-func (h *BindHandle) DropInvalidBindRecord() {
-	invalidBindRecordMap := copyInvalidBindRecordMap(h.invalidBindRecordMap.Load().(map[string]*invalidBindRecordMap))
-	for key, invalidBindRecord := range invalidBindRecordMap {
-		if invalidBindRecord.droppedTime.IsZero() {
-			err := h.DropBindRecord(invalidBindRecord.bindRecord)
+// tmpBindRecordMap is used to temporarily save bind record changes.
+// Those changes will be flushed into store periodically.
+type tmpBindRecordMap struct {
+	sync.Mutex
+	atomic.Value
+	flushFunc func(record *BindRecord) error
+}
+
+func (tmpMap *tmpBindRecordMap) flushToStore() {
+	newMap := copyBindRecordUpdateMap(tmpMap.Load().(map[string]*bindRecordUpdate))
+	for key, bindRecord := range newMap {
+		if bindRecord.updateTime.IsZero() {
+			err := tmpMap.flushFunc(bindRecord.bindRecord)
 			if err != nil {
-				logutil.BgLogger().Error("DropInvalidBindRecord failed", zap.Error(err))
+				logutil.BgLogger().Error("flush bind record failed", zap.Error(err))
 			}
-			invalidBindRecord.droppedTime = time.Now()
+			bindRecord.updateTime = time.Now()
 			continue
 		}
 
-		if time.Since(invalidBindRecord.droppedTime) > 6*time.Second {
-			delete(invalidBindRecordMap, key)
-			updateMetrics(metrics.ScopeGlobal, invalidBindRecord.bindRecord, nil, false)
+		if time.Since(bindRecord.updateTime) > 6*time.Second {
```

Comment:
What does this branch mean?

Reply:
It is used to prevent duplicated jobs. For example, one session push a job to drop a invalid record, and it succeed to update it in the cache and storeage, but other sessions may still read the old cache and re-push the same job. By keep it a little longer, we can prevent the duplicated jobs for most cases.